src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -247,6 +241,981 @@ public bool NeedsRefresh()
+                                            settingKey,
+                                            reader.TokenType.ToString(),
+                                            JsonTokenType.String.ToString());
+                                    }
+
+                                    i++;
+                                }
+
+                                featureGroupAllocation.Groups = groups;
+                            }
+                            else if (reader.TokenType != JsonTokenType.Null)
+                            {
+                                throw CreateFeatureFlagFormatException(
+                                    FeatureManagementConstants.Groups,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    JsonTokenType.StartArray.ToString());
+                            }
+
+                            break;
+                        }
+
+                    default:
+                        reader.Skip();
+
+                        break;
+                }
+            }
+
+            return featureGroupAllocation;
+        }
+
+        private FeaturePercentileAllocation ParseFeaturePercentileAllocation(ref Utf8JsonReader reader, string settingKey)
+        {
+            var featurePercentileAllocation = new FeaturePercentileAllocation();
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    continue;
+                }
+
+                string percentileAllocationPropertyName = reader.GetString();
+
+                switch (percentileAllocationPropertyName)
+                {
+                    case FeatureManagementConstants.Variant:
+                        {
+                            if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                            {
+                                featurePercentileAllocation.Variant = reader.GetString();
+                            }
+                            else if (reader.TokenType != JsonTokenType.Null)
+                            {
+                                throw CreateFeatureFlagFormatException(
+                                    FeatureManagementConstants.Variant,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    JsonTokenType.String.ToString());
+                            }
+
+                            break;
+                        }
+
+                    case FeatureManagementConstants.From:
+                        {
+                            if (reader.Read() && 
+                                ((reader.TokenType == JsonTokenType.Number && reader.TryGetInt32(out int from)) || 
```

Comment:
`from` is C# keyword.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/

How about change it to `fromPercentile`